python_modules/dagster/dagster/_serdes/serdes.py

```diff
@@ -90,6 +91,30 @@
     "UnknownSerdesValue",
 ]
 
+_K = TypeVar("_K")
+_V = TypeVar("_V")
+
+
+class SerializableNonScalarKeyMapping(Mapping[_K, _V]):
+    """Wrapper class for non-scalar key mappings, used to performantly type check when serializing
+    without having to access types of specific keys.
```

Comment:
> without having to access types of specific keys.

nit: maybe something like 

"without impacting the performance of serializing the more common scalar key dicts. May be replaceable with a different clever scheme"